feat(oas-normalize): new option to return validation errors instead of throwing (#1018)
readmeio/oas: packages/oas-normalize/README.md

Reviewer: (i know it's not related to this PR per se but i was reading the whole thing and noticed this) on line 131, i think that `validate` object needs to be nested within a `parser` object, correct? sort of like how we do here:

https://github.com/readmeio/oas/blob/f6ec6fcb02cc6de9851fc99cfde4000d9da7b0a2/packages/oas-normalize/test/index.test.ts#L571-L573

Author: will fix

}
```

> If you want to always receive a `ValidationResult` object back, which will still allow you to determine if the supplied API definition is valid or not, you can set the `throwIfInvalid` option to `false`:

Reviewer: ```suggestion
If you want to always receive a `ValidationResult` object back (which includes a flag that indicates if the supplied API definition is valid or not), you can set the `throwIfInvalid` option to `false`:
```

```ts
const result = await oas.validate({ throwIfInvalid: false });
if (result.valid) {
  if (result.warnings.length) {
    console.warn('The API definition is valid however it contains some warnings.', {
      warnings: result.warnings,
    });
  } else {
    console.log('The API definition is valid!');
  }
} else {
  console.error('The API definition is invalid.', {
    errors: result.errors,
    warnings: result.warnings,
  });
}
```

Additionally, when receiving a `ValidationResult` object back you can also transform it into a human-readable string by utilizing our `compileErrors` utility:

```ts
import { compileErrors } from 'oas-normalize/lib/utils';

const result = await oas.validate({ throwIfInvalid: false });
if (result.valid) {
  if (result.warnings.length) {

>     console.warn(compileErrors(result));

Author: actually there's better docs on that in `openapi-parser`, going to rework the note i removed and link out to those docs instead

}
```

If you want to always receive a `ValidationResult` object back (which includes a flag that indicates if the supplied API definition is valid or not), you can set the `throwIfInvalid` option to `false`:

```ts
const result = await oas.validate({ throwIfInvalid: false });
console.log(result);

> /* {

Reviewer: i slightly prefer this but don't feel strongly:

```suggestion
/*
{
```
